### PR TITLE
Migrate to React Router v6

### PR DESCRIPTION
**Pull request by mgrabovsky**
Migrate from React Router v5 to v6 in accordance with the official [migration guide][1].

[1]: https://reactrouter.com/en/6/upgrading/v5

Reference: OSCI-3354

### CODE REVIEW

**Comment by mgrabovsky** [comment]
@Andrei-Stepanov This is ready for review.

File: src/components/PageGating.tsx [hunk 1]
```diff
@@ -456,21 +451,21 @@ function BuildTypeSelector() {
             onToggle={setExpanded}
             selections={buildType}
         >
-            {_.keys(BUILD_TYPE_MENU_ITEMS).map((item, index) => (
-                <SelectOption key={index} value={item} />
+            {Object.entries(BUILD_TYPE_MENU_ITEMS).map(([label, value]) => (
```
Comment (Andrei-Stepanov) [inline src/components/PageGating.tsx:454]: Any substantial preferences over `_.keys` ?

File: src/slices/gateArtifactsSlice.ts [hunk 2]
```diff
@@ -49,18 +51,24 @@ export interface StateGatingTests {
...
+>;
+
 const INITIAL_STATE: StateGatingTests = {
-    buildType: getTypeFromSelect(query.get('btype') || 'ordinary'),
-    ciSystem: decodeURIComponent(query.get('cisystem') || ''),
-    gateTag: decodeURIComponent(query.get('gatetag') || ''),
-    ignoreCiSystem: decodeURIComponent(query.get('ic') || 'false') === 'true',
-    packager: decodeURIComponent(query.get('packager') || ''),
-    productId: _.toInteger(query.get('pid') || 604),
+    buildType: 'brew-build',
```
Comment (Andrei-Stepanov) [inline src/slices/gateArtifactsSlice.ts:64]: @mgrabovsky how do the URL args will work?

Comment (mgrabovsky) [reply src/slices/gateArtifactsSlice.ts:64]: They are handled in `PageGating` via React Router's `useSearchParams`, which is a sort of a state in itself.

**Comment by mgrabovsky** [comment]
I'm actually going to take a bit more time with this to make it even simpler.

File: src/components/PageGating.tsx (hunk 1 above)
Comment (mgrabovsky) [reply src/components/PageGating.tsx]: Changed to `_.map()`.

**Comment by mgrabovsky** [comment]
@Andrei-Stepanov I think this is ready for another review. I've deployed the build on the [dev instance](https://dashboard.dev.osci.redhat.com/#/gating) so you can check how it works live.

**Review by Andrei-Stepanov — APPROVED** [review]

**Pull request merged**